### PR TITLE
Migrate rectangle intersection demo from examples repo

### PR DESCRIPTION
**Pull request by jamwaffles**

### CODE REVIEW

File: debug-tools/examples/rectangle-intersection.rs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+                .draw(display)?;
+        }
+
+        Ok(())
+    }
+}
+
+fn main() {
+    let settings = OutputSettingsBuilder::new().scale(3).build();
+    let window = Window::new("Line debugger", &settings);
```
Comment (rfuest) [inline debug-tools/examples/rectangle-intersection.rs]: Nit:
```suggestion
    let window = Window::new("Rectangle intersection", &settings);
```

**Review by rfuest — CHANGES_REQUESTED** [review]

File: debug-tools/examples/rectangle-intersection.rs (hunk 1 above)
Comment (jamwaffles) [reply debug-tools/examples/rectangle-intersection.rs]: Good catch, thanks!

File: debug-tools/examples/rectangle-intersection.rs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            .into_styled(PrimitiveStyle::with_fill(Rgb888::GREEN))
+            .draw(display)?;
+
+        let intersection = base_rectangle.intersection(&moving_rectangle);
+
+        if intersection.size != Size::zero() {
+            intersection
+                .into_styled(PrimitiveStyle::with_fill(Rgb888::BLUE))
+                .draw(display)?;
+        }
```
Comment (rfuest) [inline debug-tools/examples/rectangle-intersection.rs]: Drawing a zero sized rectangle should be a noop anyway:
```suggestion
        intersection
            .into_styled(PrimitiveStyle::with_fill(Rgb888::BLUE))
            .draw(display)?;
```
But if you want to keep the check for some reason, the preferred way to test for zero sizes is:
```suggestion
        if intersection.is_zero_sized() {
            intersection
                .into_styled(PrimitiveStyle::with_fill(Rgb888::BLUE))
                .draw(display)?;
        }
```

**Review by rfuest — CHANGES_REQUESTED** [review]

File: debug-tools/examples/rectangle-intersection.rs (hunk 2 above)
Comment (jamwaffles) [reply debug-tools/examples/rectangle-intersection.rs]: Ah yes, it'll be a noop won't it. I've removed the check.

**Review by rfuest — APPROVED** [review]

**Pull request merged**